In [1]:
from sqlalchemy import create_engine, inspect
import pandas
dbh = create_engine('sqlite:///WF305900_data.sqlite')

In [2]:
import sqlite3

In [3]:
con = sqlite3.connect("WF305900_data.sqlite")
cursor = con.cursor()
#cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
print(cursor.fetchall())


[]


The table has these many columns of which "pepprotmap" gives the mapping file? If so how to reteive the same

In [5]:
cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
print(cursor.fetchall())

[('pclitrans',), ('pcliprot',), ('annotations',), ('nbunit',), ('fasta',), ('traml',), ('reqfeat',), ('matrix',), ('pepprotmap',)]


In [6]:
cursor.execute("SELECT * FROM pepprotmap;")
print(cursor.fetchall())

[('LDTMDELKEK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('MTENEQIFWNR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('MTVIPIEEIQAQVGK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('NLDLLLIDDIQSLAK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('NMISQDESLR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('QATYEFFVHDAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('QIVLTSDR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('QQALNSLPTVTSDLNSK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('QVAMFLAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('THLLNAIGNSVLLENPNAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('TLSGTQEEFFNTFNALHNNNK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('TPDHLNDLEDR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('TQNIVLAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), ('VLELAQSQLK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|')

"LDTMDELKEK" is the peptide .What is target and What is 'fid|19734402|locus|VBIStrPyo96585_0001|'
So basically the term VBIStrPy96585 remains same and others change and there are 2013 ids found
What does this mean?

For example fid 19734402_0001 has this many peptides
('LDTMDELKEK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('MTENEQIFWNR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('MTVIPIEEIQAQVGK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('NLDLLLIDDIQSLAK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('NMISQDESLR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('QATYEFFVHDAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('QIVLTSDR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('QQALNSLPTVTSDLNSK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('QVAMFLAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('THLLNAIGNSVLLENPNAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('TLSGTQEEFFNTFNALHNNNK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('TPDHLNDLEDR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('TQNIVLAR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('VLELAQSQLK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('WGLTVNITPPDFETR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('YITAENFINEFVIHIR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('YSFENFIQGDENR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('FYGVTVK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('HIATIYLDQMK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'), 
('IEIETIK', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|'),
('IQEYNFIFPQDTIEYLAGQFDSNVR', 'target', 'fid|19734402|locus|VBIStrPyo96585_0001|')

fId 19734404_0002 has this many and so on and so forth
('LEITQNHISAHVNSPEVGK', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'), 
('LLEGNYPDTDR', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'), 
('LQEVSTENPLILK', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'),
('SEHISFYTR', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'), 
('DVDQYPR', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'), 
('EIEQHQVVLTSGK', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'), 
('AFLISNATQNGTVK', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'),
('IHFLSPVRPFTLTPGDEEESFIQLITPVR', 'target', 'fid|19734404|locus|VBIStrPyo96585_0002|'),

What is pclitrans and pcliprot and what does this number denote? How are these two different

In [7]:
cursor.execute("SELECT * FROM pclitrans;")
print(cursor.fetchall())

[(0.0, 0.012928, 1e-06, 9113, 1, 585424, 659462, 0.000113, 0.013630000000000001, 4.943773), (0.01, 0.817345, 0.009429, 546457, 5520, 579905, 122118, 0.01, 0.817346, 2.351589), (0.02, 0.872735, 0.020340999999999998, 583490, 11908, 573517, 85085, 0.02, 0.8727370000000001, 1.9277220000000002), (0.05, 0.9299209999999999, 0.055899000000000004, 621717, 32725, 552700, 46858, 0.050004, 0.9299139999999999, 1.413241), (0.1, 0.965462, 0.12250699999999999, 645468, 71719, 513706, 23107, 0.1, 0.9654379999999999, 1.016657), (0.2, 0.992685, 0.28341900000000003, 663681, 165921, 419504, 4894, 0.2, 0.9926799999999999, 0.5226029999999999), (0.3, 1.0, 0.49126899999999996, 671069, 287601, 297824, -2494, 0.3, 1.0, 0.073914), (0.4, 1.0, 0.769027, 675310, 450208, 135217, -6735, 0.400001, 1.0, -0.60035), (0.5, 1.0, 1.0, 668575, 585425, 0, 0, 0.466846, 1.0, -565.5306400000001)]


In [8]:
cursor.execute("SELECT * FROM pcliprot;")
print(cursor.fetchall())

[(0.0, 0.016178, 2e-06, 8896, 1, 444639, 513182, 0.00010800000000000001, 0.01704, 4.943779), (0.01, 0.819469, 0.009718000000000001, 427828, 4321, 440319, 94250, 0.009999, 0.819471, 2.423031), (0.02, 0.8723639999999999, 0.020904, 455443, 9295, 435345, 66635, 0.02, 0.872366, 1.996593), (0.05, 0.927873, 0.057340999999999996, 484421, 25496, 419144, 37657, 0.05, 0.9278709999999999, 1.480326), (0.1, 0.963139, 0.125653, 502835, 55870, 388770, 19243, 0.1, 0.963142, 1.0808229999999999), (0.2, 0.9913129999999999, 0.290983, 517530, 129383, 315257, 4548, 0.2, 0.991289, 0.587696), (0.3, 1.0, 0.505506, 524465, 224768, 219872, -2387, 0.299998, 1.0, 0.133154), (0.4, 1.0, 0.791918, 528175, 352118, 92522, -6097, 0.400001, 1.0, -0.583469), (0.5, 1.0, 1.0, 522078, 444640, 0, 0, 0.45994799999999997, 1.0, -565.5306400000001)]


In [9]:
cursor.execute("SELECT * FROM annotations;")
print(cursor.fetchall())

[(1, 'MICROORGANISM', 'NA', 'NONE', 'NA', 'NONE', 'LABELFREE', '4', '0', '0', 'JM-M3-10', 'NA', 'NONE', 'MGAS19979', '1314', 'BIOLGROUP', 'GROUP', 'GENOME', 'non-invasive', '1', 'SRR125478', 'NA', 'false', 'false', 'ProteoWizard', 'PP_S1701_035.mzML.gz', '1023049161', '94565fd675b4b427746ca4ce6f36d64c', 'fail', 'false', 'DIA', '- (no instrument method)', None, '/MS_DATA/PP_S1701_035', '20170120163714259-1107957'), (2, 'MICROORGANISM', 'NA', 'NONE', 'NA', 'NONE', 'LABELFREE', '2', '0', '0', 'JM-M3-42', 'NA', 'NONE', 'MGAS9913', '1314', 'BIOLGROUP', 'GROUP', 'GENOME', 'invasive', '1', 'SRR016727', 'NA', 'false', 'false', 'ProteoWizard', 'PP_S1701_036.mzML.gz', '1044151745', 'dbc3ed812e97e2d7ec53ac621b4864e4', 'fail', 'false', 'DIA', '- (no instrument method)', None, '/MS_DATA/PP_S1701_036', '20170125090719367-1108112'), (3, 'MICROORGANISM', 'NA', 'NONE', 'NA', 'NONE', 'LABELFREE', '4', '0', '0', 'JM-M3-11', 'NA', 'NONE', 'MGAS20112', '1314', 'BIOLGROUP', 'GROUP', 'GENOME', 'non-invasive'

Just to check the parameters in each of the rows placed one below the other

(1, 'MICROORGANISM', 'NA', 'NONE', 'NA', 'NONE', 'LABELFREE', '4', '0', '0', 'JM-M3-10', 'NA', 'NONE', 'MGAS19979', '1314', 'BIOLGROUP', 'GROUP', 'GENOME', 'non-invasive', '1', 'SRR125478', 'NA', 'false', 'false', 'ProteoWizard', 'PP_S1701_035.mzML.gz', '1023049161', '94565fd675b4b427746ca4ce6f36d64c', 'fail', 'false', 'DIA', '- (no instrument method)', None, '/MS_DATA/PP_S1701_035', '20170120163714259-1107957'),

(2, 'MICROORGANISM', 'NA', 'NONE', 'NA', 'NONE', 'LABELFREE', '2', '0', '0', 'JM-M3-42', 'NA', 'NONE', 'MGAS9913', '1314', 'BIOLGROUP', 'GROUP', 'GENOME', 'invasive', '1', 'SRR016727', 'NA', 'false', 'false', 'ProteoWizard', 'PP_S1701_036.mzML.gz', '1044151745', 'dbc3ed812e97e2d7ec53ac621b4864e4', 'fail', 'false', 'DIA', '- (no instrument method)', None, '/MS_DATA/PP_S1701_036', '20170125090719367-1108112'), 

In [10]:
cursor.execute("SELECT * FROM nbunit;")
print(cursor.fetchall())

[(1, 'report', 1, 'R', '### connect to the database\nneeded in all notebooks to get access to data', "library(RSQLite)\ndbh <- dbConnect(SQLite(),db='data.sqlite')\ndbListTables(dbh)", ''), (2, 'report', 1, 'R', '### Additional reports\n- [qc.ipynb](qc.ipynb)\n- [quant.ipynb](quant.ipynb)\n- [py.ipynb](py.ipynb)', '', ''), (3, 'report', 1, 'R', '### experiment report\n- design and more\n- demo the annotations table', 'row <- dbGetQuery(dbh,\\"SELECT * FROM annotations WHERE id = 1\\")\nt(row)\ndbGetQuery(dbh,\\"SELECT id,permid,ds_file_name,bs_name,bs_treatment_type1,bs_treatment_value1 FROM annotations\\")', ''), (4, 'quant', 1, 'R', '### connect to the database\nneeded in all notebooks to get access to data', "library(RSQLite)\ndbh <- dbConnect(SQLite(),db='data.sqlite')\ndbListTables(dbh)", ''), (5, 'quant', 1, 'R', '### pepprotmap:\n', 'dbGetQuery(dbh,\\"SELECT \'n target peptides not supported by protein fasta file\' as description,COUNT(*) \nFROM matrix WHERE pep NOT IN (select p

This is the fasta file and is this our "AREA OF INTEREST"?

In [11]:
cursor.execute("SELECT * FROM fasta;")
print(cursor.fetchall())

[('fid|19734402|locus|VBIStrPyo96585_0001|   Chromosomal replication initiator protein DnaA   [Streptococcus pyogenes MGAS315]', 'fid|19734402|locus|VBIStrPyo96585_0001|', 'MTENEQIFWNRVLELAQSQLKQATYEFFVHDARLLKVDKHIATIYLDQMKELFWEKNLKDVILTAGFEVYNAQISVDYVFEEDLMIEQNQTKINQKPKQQALNSLPTVTSDLNSKYSFENFIQGDENRWAVAASIAVANTPGTTYNPLFIWGGPGLGKTHLLNAIGNSVLLENPNARIKYITAENFINEFVIHIRLDTMDELKEKFRNLDLLLIDDIQSLAKKTLSGTQEEFFNTFNALHNNNKQIVLTSDRTPDHLNDLEDRLVTRFKWGLTVNITPPDFETRVAILTNKIQEYNFIFPQDTIEYLAGQFDSNVRDLEGALKDISLVANFKQIDTITVDIAAEAIRARKQDGPKMTVIPIEEIQAQVGKFYGVTVKEIKATKRTQNIVLARQVAMFLAREMTDNSLPKIGKEFGGRDHSTVLHAYNKIKNMISQDESLRIEIETIKNKIK'), ('fid|19734404|locus|VBIStrPyo96585_0002|   DNA polymerase III beta subunit (EC 2.7.7.7)   [Streptococcus pyogenes MGAS315]', 'fid|19734404|locus|VBIStrPyo96585_0002|', 'MIQFSINRTLFIHALNATKRAISTKNAIPILSSIKIEVTPTGVTLTGSNGQISIENTIPVSNENAGLLITSPGAILLEASFFINIISSLPDININVKEIEQHQVVLTSGKSEITLKGKDVDQYPRLQEVSTENPLILKTKLLKSIIAETAFAASLQESRPILTGVHIVLSNHKDFKAVATDSHRMSQRLITLDNTSADFDVV

In [12]:
cursor.execute("SELECT * FROM traml;")
print(cursor.fetchall())

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [13]:
cursor.execute("SELECT * FROM reqfeat;")
print(cursor.fetchall())

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [14]:
cursor.execute("SELECT * FROM matrix ;")
print(cursor.fetchall())

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



These are experimental parameters and how important are  these from the Bioinfo parse?msi_children,dda_children,dia_children,primary_cell_type,sample_preparation,strain_name.....

In [15]:
cursor.execute("SELECT sql FROM sqlite_master WHERE name = 'annotations';")
print(cursor.fetchall())

[('CREATE TABLE annotations (\n\tid INTEGER NOT NULL, \n\tbs_biological_sample_type VARCHAR, \n\tbs_ck_patients VARCHAR, \n\tbs_digestion VARCHAR, \n\tbs_em_patients VARCHAR, \n\tbs_fractionation VARCHAR, \n\tbs_labeling VARCHAR, \n\tbs_msi_children VARCHAR, \n\tbs_msi_dda_children VARCHAR, \n\tbs_msi_dia_children VARCHAR, \n\tbs_name VARCHAR, \n\tbs_primary_cell_type VARCHAR, \n\tbs_sample_preparation VARCHAR, \n\tbs_strain_name VARCHAR, \n\tbs_tax_id VARCHAR, \n\tbs_treatment_type1 VARCHAR, \n\tbs_treatment_type2 VARCHAR, \n\tbs_treatment_type3 VARCHAR, \n\tbs_treatment_value1 VARCHAR, \n\tbs_treatment_value2 VARCHAR, \n\tbs_treatment_value3 VARCHAR, \n\tds_acquisition_type VARCHAR, \n\tds_centroided VARCHAR, \n\tds_chromatogram VARCHAR, \n\tds_conversion_software VARCHAR, \n\tds_file_name VARCHAR, \n\tds_file_size VARCHAR, \n\tds_md5_checksum VARCHAR, \n\tds_sample_acquisition_type VARCHAR, \n\tds_sim_as_spectra VARCHAR, \n\tmsi_acquisition_type VARCHAR, \n\tmsi_transitions VARCHAR,

There are 64 strains 

In [16]:
cursor.execute("SELECT bs_strain_name FROM annotations;")
print(cursor.fetchall())

[('MGAS19979',), ('MGAS9913',), ('MGAS20112',), ('MGAS9983',), ('MGAS20337',), ('MGAS9969',), ('MGAS19464',), ('MGAS15038',), ('MGAS20333',), ('MGAS15003',), ('MGAS20326',), ('MGAS9901',), ('MGAS16608',), ('MGAS15194',), ('MGAS16682',), ('MGAS15010',), ('MGAS16453',), ('MGAS15025',), ('MGAS16616',), ('MGAS15192',), ('MGAS17651',), ('MGAS16474',), ('MGAS19972',), ('MGAS15052',), ('MGAS16029',), ('MGAS10839',), ('MGAS19549',), ('MGAS15054',), ('MGAS16775',), ('MGAS15061',), ('MGAS18726',), ('MGAS9914',), ('MGAS17763',), ('MGAS16479',), ('MGAS19429',), ('MGAS15009',), ('MGAS16946',), ('MGAS15187',), ('MGAS19445',), ('MGAS9924',), ('MGAS16407',), ('MGAS15062',), ('MGAS19786',), ('MGAS9956',), ('MGAS19459',), ('MGAS9907',), ('MGAS19771',), ('MGAS10833',), ('MGAS17538',), ('MGAS15041',), ('MGAS19510',), ('MGAS9952',), ('MGAS17066',), ('MGAS16477',), ('MGAS16879',), ('MGAS15013',), ('MGAS19478',), ('MGAS9936',), ('MGAS19511',), ('MGAS9928',), ('MGAS16628',), ('MGAS15032',), ('MGAS19646',), ('

This is the matrix which I had and was doing all Data Analyis Heatmap ,Umap generated

In [17]:
cursor.execute("SELECT sql FROM sqlite_master WHERE name = 'matrix';")
print(cursor.fetchall())


[('CREATE TABLE matrix (\n\t"Peptide" TEXT, \n\t"Protein" TEXT, \n\t"Intensity_PP_S1701_078_scored.txt_0_0" FLOAT, \n\t"RT_PP_S1701_078_scored.txt_0_0" FLOAT, \n\t"score_PP_S1701_078_scored.txt_0_0" FLOAT, \n\t"Intensity_PP_S1701_083_scored.txt_0_1" FLOAT, \n\t"RT_PP_S1701_083_scored.txt_0_1" FLOAT, \n\t"score_PP_S1701_083_scored.txt_0_1" FLOAT, \n\t"Intensity_PP_S1701_056_scored.txt_0_2" FLOAT, \n\t"RT_PP_S1701_056_scored.txt_0_2" FLOAT, \n\t"score_PP_S1701_056_scored.txt_0_2" FLOAT, \n\t"Intensity_PP_S1701_038_scored.txt_0_3" FLOAT, \n\t"RT_PP_S1701_038_scored.txt_0_3" FLOAT, \n\t"score_PP_S1701_038_scored.txt_0_3" FLOAT, \n\t"Intensity_PP_S1701_074_scored.txt_0_4" FLOAT, \n\t"RT_PP_S1701_074_scored.txt_0_4" FLOAT, \n\t"score_PP_S1701_074_scored.txt_0_4" FLOAT, \n\t"Intensity_PP_S1701_046_scored.txt_0_5" FLOAT, \n\t"RT_PP_S1701_046_scored.txt_0_5" FLOAT, \n\t"score_PP_S1701_046_scored.txt_0_5" FLOAT, \n\t"Intensity_PP_S1701_077_scored.txt_0_6" FLOAT, \n\t"RT_PP_S1701_077_scored.txt_0